consensus/state_sync/docs/State_Sync.md

```diff
@@ -0,0 +1,7 @@
+## State Sync Lifecycle
```

Comment:
We already have `consensus/doc/PROTOCOL_STATE_SYNC.md`.

Can you update that doc appropriate rather than creating a new one?